src/Functions/FunctionsStringArray.h

```diff
@@ -505,6 +715,39 @@ class SplitByRegexpImpl
 
             token_begin = pos;
 
+            if (max_splits)
+            {
+                switch (max_substring_behavior)
+                {
+                    case MaxSubstringBehavior::LikeClickHouse:
+                    {
+                        if (splits == *max_splits)
+                            return false;
+                        break;
+                    }
+                    case MaxSubstringBehavior::LikeSpark:
+                    {
+                        if (splits == *max_splits - 1)
+                        {
+                            token_end = end;
+                            pos = nullptr;
+                            return true;
+                        }
+                        break;
+                    }
+                    case MaxSubstringBehavior::LikePython:
+                    {
+                        if (splits == *max_splits)
+                        {
+                            token_end = end;
+                            pos = nullptr;
+                            return true;
+                        }
+                        break;
+                    }
+                }
```

Comment:
similar chunk seems to be repeated often

Reply:
This will hopefully become simpler.

Reply:
There are some subtle differences between places where the chunk is used. Can't pull it out easily. It is much simpler in the last commit, guess we can keep it as is.